File: JS/01.Basics/02.3-ConditionalStatementsExercise/06.WorldSwimmingRecord.js
function swimmingRecord(data) {
    
    let record = Number(data[0]);
    let distance = Number(data[1]);
    let timePerMeter = Number(data[2]);

    
    record = parseFloat(record);
    distance = parseFloat(distance);
    timePerMeter = parseFloat(timePerMeter);

    let resistance = Math.floor(distance / 15) * 12.5;
    let totalTime = (distance * timePerMeter) + resistance;

    if (totalTime < record) {
        console.log(`Yes, he succeeded! The new world record is ${totalTime.toFixed(2)} seconds.`);
    } else {
        let timeDifference = totalTime - record;
        console.log(`No, he failed! He was ${timeDifference.toFixed(2)} seconds slower.`);
    }
}

swimmingRecord((["10464",
"1500",
"20"])
);
swimmingRecord((["55555.67",
"3017",
"5.03"])
);

File: C++/01-Basics/02.3-ConditionalStatementsExercise/06.WorldSwimmingRecord/06.WorldSwimmingRecord.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

int main() {
    double recordSeconds, distance, timePerMeter;

    cin >> recordSeconds >> distance >> timePerMeter;

    double resistanceTime = floor(distance / 15) * 12.5;

    double ivanTime = distance * timePerMeter + resistanceTime;

    cout << fixed << setprecision(2);

    if (ivanTime < recordSeconds) {
        cout << "Yes, he succeeded! The new world record is " << ivanTime << " seconds." << endl;
    }
    else {
        cout << "No, he failed! He was " << ivanTime - recordSeconds << " seconds slower." << endl;
    }

    return 0;
}
